Tests incompatible with Git<2.28

I tried to create a dev setup to contribute. (Sadly https://github.com/akaihola/darker/blob/master/CONTRIBUTING.rst doesn't contain how to get started)

Running tests i get this error:
```
    def check_call(*popenargs, **kwargs):
        """Run command with arguments.  Wait for command to complete.  If
        the exit code was zero then return, otherwise raise
        CalledProcessError.  The CalledProcessError object will have the
        return code in the returncode attribute.
    
        The arguments are the same as for the call function.  Example:
    
        check_call(["ls", "-l"])
        """
        retcode = call(*popenargs, **kwargs)
        if retcode:
            cmd = kwargs.get("args")
            if cmd is None:
                cmd = popenargs[0]
>           raise CalledProcessError(retcode, cmd)
E           subprocess.CalledProcessError: Command '['git', 'init', '--initial-branch=master']' returned non-zero exit status 129.

/usr/lib/python3.9/subprocess.py:373: CalledProcessError
---------------------------- Captured stderr setup -----------------------------
error: unknown option `initial-branch=master'
usage: git init [-q | --quiet] [--bare] [--template=<template-directory>] [--shared[=<permissions>]] [<directory>]

    --template <template-directory>
                          directory from which templates will be used
    --bare                create a bare repository
    --shared[=<permissions>]
                          specify that the git repository is to be shared amongst several users
    -q, --quiet           be quiet
    --separate-git-dir <gitdir>
                          separate git dir from working tree
```
I have `git --version` is: `2.25.1`

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -13,6 +13,7 @@ Fixed
   Darker's own code base.
 - Fix compatibility with ``black-22.10.1.dev19+gffaaf48`` and later – an argument was
   replaced in ``black.files.gen_python_files()``.
+- Fix tests to work with Git older than version 2.28.x.
 
 
 1.6.0_ - 2022-12-19

--- a/src/darker/git.py
+++ b/src/darker/git.py
@@ -9,7 +9,18 @@ from datetime import datetime
 from functools import lru_cache
 from pathlib import Path
 from subprocess import DEVNULL, PIPE, CalledProcessError, check_output, run  # nosec
-from typing import Dict, Iterable, List, Optional, Set, Tuple, Union, overload
+from typing import (
+    Dict,
+    Iterable,
+    List,
+    Match,
+    Optional,
+    Set,
+    Tuple,
+    Union,
+    cast,
+    overload,
+)
 
 from darker.diff import diff_and_get_opcodes, opcodes_to_edit_linenums
 from darker.multiline_strings import get_multiline_string_ranges
@@ -31,6 +42,28 @@ COMMIT_RANGE_RE = re.compile(r"(.*?)(\.{2,3})(.*)$")
 #   for determining the revision range
 WORKTREE = ":WORKTREE:"
 PRE_COMMIT_FROM_TO_REFS = ":PRE-COMMIT:"
+
+
+def git_get_version() -> Tuple[int, ...]:
+    """Return the Git version as a tuple of integers
+
+    Ignores any suffixes to the dot-separated parts of the version string.
+
+    :return: The version number of Git installed on the system
+    :raise: ``RuntimeError`` if unable to parse the Git version
+
+    """
+    output_lines = _git_check_output_lines(["--version"], Path("."))
+    version_string = output_lines[0].rsplit(None, 1)[-1]
+    # The version string might be e.g.
+    # - "2.39.0.windows.1"
+    # - "2.36.2"
+    part_matches = [re.match(r"\d+", part) for part in version_string.split(".")][:3]
+    if all(part_matches):
+        return tuple(
+            int(match.group(0)) for match in cast(List[Match[str]], part_matches)
+        )
+    raise RuntimeError(f"Unable to parse Git version: {output_lines!r}")
 
 
 def git_is_repository(path: Path) -> bool:

--- a/src/darker/tests/conftest.py
+++ b/src/darker/tests/conftest.py
@@ -9,7 +9,7 @@ from typing import Dict, Iterable, List, Union
 import pytest
 from black import find_project_root as black_find_project_root
 
-from darker.git import _git_check_output_lines
+from darker.git import _git_check_output_lines, git_get_version
 
 
 class GitRepoFixture:
@@ -27,7 +27,10 @@ class GitRepoFixture:
         env = {"HOME": str(root), "LC_ALL": "C", "PATH": os.environ["PATH"]}
         instance = cls(root, env)
         # pylint: disable=protected-access
-        instance._run("init", "--initial-branch=master")
+        force_master = (
+            ["--initial-branch=master"] if git_get_version() >= (2, 28) else []
+        )
+        instance._run("init", *force_master)
         instance._run("config", "user.email", "ci@example.com")
         instance._run("config", "user.name", "CI system")
         return instance

